### DISCUSSION TITLE
Issues with datafiles in the wrong directory.

### DISCUSSION THREAD

**Discussion by sorenwacker** [pyinstaller/pyinstaller#7474] [Help]
https://github.com/pymzml/pymzML/issues/324

I am trying to use `pymzml` with pyinstaller, and managed to do so before. I recently switched to `pyinstaller=5.8.0`. And now I am running into this issue with some of the datafiles. For some reason the files are put into the wrong directoy:

`FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\asdf\\workspace\\ms-mint-app\\pyinstaller\\dist\\Mint\\obo\\psi-ms-1.1.0.obo'`

Instead of 

`FileNotFoundError: [Errno 2] No such file or directory: 'C:\\Users\\asdf\\workspace\\ms-mint-app\\pyinstaller\\dist\\Mint\\pymzml\\obo\\psi-ms-1.1.0.obo'`

My specfile is here:

`https://github.com/LewisResearchGroup/ms-mint-app
`
in the `pyinstaller` directory.

I tried copying the files to the expected location, but then I am running into another issue on windows:

I would like to understand why the installed library expects the files at the wrong path and how to fix it.

**Comment by rokm** [comment]
It's because [`pymzml` adjusts the path when ran from frozen application](https://github.com/pymzml/pymzML/blob/2a546d0cec9e80d9b4bad2a9640db4898cd89a56/pymzml/obo.py#L174-L178). 

The comment suggests that this is done for compatibility with `cx_freeze`; but this breaks PyInstaller, which collects the data files with original sub-directory layout (`pymzml/obo/*.obo`).

**Reply by rokm** [reply]
So I suppose if `pymzml` needs special path for `cx_freeze`, they should modify that frozen-application branch to first check whether `os.path.join(os.path.dirname(__file__), 'obo')` exists (= PyInstaller) before falling back to `os.path.join(os.path.dirname(sys.executable), 'obo')` (= cx_freeze).
